2006-01-02  Samuel Thibault <samuel.thibault@ens-lyon.org>

	* i386/i386at/iopl.c (iopl_port_list): Add timer controler port.
	* i386/i386at/kd.c (vga_port_list): Renamed to ...
	(kd_port_list): ... this. Add timer and speaker ports.


Index: i386/i386at/iopl.c
===================================================================
--- i386/i386at/iopl.c.orig	2006-10-26 00:16:50.000000000 +0300
+++ i386/i386at/iopl.c	2006-10-26 00:17:28.000000000 +0300
@@ -47,7 +47,7 @@ mach_device_t	iopl_device = 0;
  */
 io_reg_t iopl_port_list[] = {
 	/* timer 2 */
-	0x42,
+	0x42, 0x43,
 	/* speaker output */
 	0x61,
 	/* ATI - savage */
Index: i386/i386at/kd.c
===================================================================
--- i386/i386at/kd.c.orig	2006-10-26 00:16:50.000000000 +0300
+++ i386/i386at/kd.c	2006-10-26 00:17:28.000000000 +0300
@@ -331,7 +331,9 @@ short	kd_io_reg	= EGA_IO_REG;
 /*
  * IO port sets for different controllers.
  */
-io_reg_t vga_port_list[] = {
+static io_reg_t kd_port_list[] = {
+	0x42, 0x43,	/* Timer */
+	0x61,		/* Speaker */
 	0x3b4, 0x3b5, 0x3b8, 0x3b9, 0x3ba,	/* MDA/EGA */
 	0x3d4, 0x3d5, 0x3d8, 0x3d9, 0x3da,	/* CGA/EGA */
 	0x3c0, 0x3c1, 0x3c2, 0x3c3, 0x3c4, 0x3c5, 0x3c6, 0x3c7,
@@ -345,7 +347,7 @@ kd_io_map_open(device)
 	mach_device_t	device;
 {
 	kd_io_device = device;
-	io_port_create(device, vga_port_list);
+	io_port_create(device, kd_port_list);
 }
 
 kd_io_map_close()
